src/main/java/com/iexec/commons/poco/task/TaskDescription.java

```diff
@@ -162,8 +162,7 @@ public String getAppCommand() {
      */
     public boolean isEligibleToContributeAndFinalize() {
         return isTeeTask
-                && BigInteger.ONE.equals(trust)
-                && !containsCallback();
+                && BigInteger.ONE.equals(trust);
```

Comment:
This could become a one liner
`return isTeeTask && BigInteger.ONE.equals(trust);`